src/payment/strategies/braintree/braintree-payment-processor.ts

```diff
@@ -150,6 +150,9 @@ export default class BraintreePaymentProcessor {
                 amount,
                 nonce,
                 removeFrame,
+                onLookupComplete: (_data, next) => {
```

Comment:
It appears that `onLookupComplete` is a deprecated option according to the [doc](https://braintree.github.io/braintree-web/current/ThreeDSecure.html#verifyCard). It suggests us to do something like `threeDSecure.on('lookup-complete')` instead.

I also notice that there are a couple of other [deprecated options](https://developers.braintreepayments.com/guides/3d-secure/migration/javascript/v3#can-merchants-customize-the-presentation-of-the-bank-challenge?) that are only relevant for 3DS1, i.e.: `addFrame` and `removeFrame` because, with 3DS2, the presentation of the challenge is controlled by the issuing bank. Perhaps they don't need to be passed if we're migrating everyone to v2.

Another question I have is backwards compatibility. Is this upgrade a backwards compatible upgrade?

Reply:
Thanks for pointing to me to their SDK reference. I see that couple of their options in it shown as deprecated. Thay obviously have a documentation inconsistency :(
I've made changes according to https://developers.braintreepayments.com/guides/3d-secure/migration/javascript/v3 guide and tested it with provided test cards. All works as expected:
both v1 and v2 test cases are working, so to be honest, i don't see any reason to change something else radically from this point. 
 
> Is this upgrade a backwards compatible upgrade? 

If I understood the question correctly, yes.
`3DS 2 works on top of your existing 3DS 1 integration. The SDK will fall back to 3DS 1 when 3DS 2 is unavailable for the card being verified.`